### DISCUSSION TITLE
Which tree is a site in?

### DISCUSSION THREAD

**Discussion by hyanwong** [tskit-dev/tskit#1546] [Q&A] [answered]
In case anyone else has the same question, I assume this is the best way to find the tree for a given site:

```
tree_index = np.searchsorted(ts.breakpoints(as_array=True), ts.site(6).position, side="right") - 1
```

The `side="right"` and `- 1` are not immediately obvious, I think? So worth having here as a recipe, or even having a `Site.tree()` or perhaps `Site.tree_index()` method?

(addendum: I'm assuming this is far more efficient than `ts.at(ts.site(6).position).id`?)

**Comment by jeromekelleher** [comment, marked as answer]
Yes, that's the best way to get the tree index all right. We could add a ``Site.tree_index()`` method I guess.
